src/Http/Controllers/VerificationController.php

```diff
@@ -10,28 +10,13 @@ class VerificationController extends Controller
 {
     public function index(VerificationRequest $request)
     {
-        $orders = collect(Order::all())
-            ->filter(function ($order) {
-                return $order->get('is_paid') === true;
-            })
-            ->map(function ($order) use ($request) {
-                foreach ($order->get('items') as $item) {
-                    if (isset($item['metadata']['license_key']) && $item['metadata']['license_key'] === $request->license_key) {
-                        return ['result' => true];
-                    }
-                }
-
-                return ['result' => false];
-            })
-            ->where('result', true)
-            ->flatten()
-            ->toArray();
-
-        if (isset($orders[0])) {
-            return $this->validResponse($request);
-        }
-
-        return $this->invalidResponse($request);
+        $order = Order::query()
+            ->where('is_paid', '=', 'true') // make sure the order is paid
+            ->where('items->metadata->license_key', '=', $request->license_key)
```

Comment:
This won't work for flat-file orders, since they need to be queried differently, see how something similar is done in Simple Commerce itself:

https://github.com/duncanmcclean/simple-commerce/blob/b77702c430acf6156d913508215c8d8b1dc9fc16/src/Query/Scopes/OrderContainsProduct.php#L41-L50

Reply:
We also don't need the `'='` here - it'll default to being an equals.

Reply:
@duncanmcclean thanks! couple questions based on the linked code:

1. as `is_paid` isn't used in v5, does that mean the unit test is broken?
2. The linked code seems to have a bunch of TODOs in them related to Eloquent. Does that mean it does *not* support a database and only flat files? 
3. Does that mean an order can only have 10 products? 

Thanks

Reply:
1. Yes, I've fixed it in https://github.com/duncanmcclean/sc-digital-products/pull/127 which will be included in the next release.
2. Yes, the code I linked is a Control Panel filter that currently only supports flat files. I can't quite remember why the Eloquent if statement has been blank.
3. Yes, in that code, it'll only check the first 10 products because there's no way of querying the Stache with a wildcard. There was statamic/cms#7454 which has been closed for the time being.